### ISSUE TITLE
Add report button to articles

### ISSUE THREAD

**Issue by NiallJoeMaher** [codu-code/codu#556]
## Context

An extension of #545, we should be able to report blog posts. 

We should add a "Report post" option to the side menu:
 
<img width="1096" alt="Showing the open options button" src="https://github.com/codu-code/codu/assets/12615742/d4d35e20-7b31-4ed4-888d-e6d525254a90">

## Expected Behavior

We should open a report modal and send an email similar to the report comment functionality.

## Current Behavior

No report buttons for posts

**Comment by NiallJoeMaher** [comment]
Giving @Fishamble dibs on this one.

**Comment by John-Paul-Larkin** [comment]
Cool, i will take this. Thanks.

**Comment by John-Paul-Larkin** [comment]
So, I think I'm learning the importance of planning!
The menu component(@headlessui/react) on the options bar contains href links. I have added a button to the menu to open the ReportComments component. I styled it to match, but I don't think that is ideal?

I built the ReportComments component to include the flag button, rather than just the modal. My idea at the time was to keep all the logic encapsulated. I have added state from the menu component to show or hide the flag, which again is not ideal. It works, but It is all a bit messy for my liking. 

Any thoughts on this? 

https://youtu.be/Wp1MfZ_c2sc
